### PR TITLE
simpler unique diag directories for //

### PR DESCRIPTION
**Pull request by PhilipDeegan**
contains https://github.com/PHAREHUB/PHARE/pull/764

simpler unique diag directories for //

## Summary by CodeRabbit

- New Feature: Added `meshgrid` method to the `PatchData` class, enhancing data visualization capabilities by generating a mesh grid based on data dimensions.
- Refactor: Simplified and improved readability of test methods by reordering arguments and removing unused variables in `getHierarchy` function calls across multiple test files.
- Test: Enhanced test coverage by modifying existing test methods to accept additional parameters and arguments, ensuring more robust testing scenarios.
- Chore: Introduced a new method `unique_diag_dir_for_test_case` for generating unique directory paths for diagnostic outputs in tests, improving test data organization.

### CODE REVIEW

File: tests/simulator/initialize/test_fields_init_2d.py [hunk 1]
```diff
@@ -34,10 +34,7 @@ def test_density_is_as_provided_by_user(self, interp_order):
         print(f"\n{self._testMethodName}_{ndim}d")
         self._test_density_is_as_provided_by_user(ndim, interp_order)
 
-    @data(*interp_orders)
-    def test_density_decreases_as_1overSqrtN(self, interp_order):
-        print(f"\n{self._testMethodName}_{ndim}d")
-        self._test_density_decreases_as_1overSqrtN(ndim, interp_order)
+
```
Comment (nicolasaunai) [inline tests/simulator/initialize/test_fields_init_2d.py:37]: what's the reason already, taking too long?

Comment (PhilipDeegan) [reply tests/simulator/initialize/test_fields_init_2d.py]: it was only ever setup for 1d, and the 2d test just ran it in 1d again

File: tests/simulator/test_initialization.py [hunk 2]
```diff
@@ -328,13 +326,12 @@ def _test_B_is_as_provided_by_user(self, dim, interp_order, **kwargs):
 
     def _test_bulkvel_is_as_provided_by_user(self, dim, interp_order):
         hier = self.getHierarchy(
+            dim,
             interp_order,
             {"L0": {"B0": nDBox(dim, 10, 19)}},
             "moments",
-            nbr_part_per_cell=100,
+            nbr_part_per_cell=100,  # ppc needs to be 10000?
```
Comment (nicolasaunai) [inline tests/simulator/test_initialization.py:333]: question?

File: tests/simulator/test_initialization.py [hunk 3]
```diff
@@ -603,21 +602,19 @@ def _test_density_decreases_as_1overSqrtN(self, dim, interp_order):
         plt.xlabel("nbr_particles")
         plt.legend()
         plt.savefig(
-            "noise_nppc_minus_theory_interp_{}_{}.png".format(dim, interp_order)
+            f"noise_nppc_minus_theory_interp_{dim}_{interp_order}.png"
```
Comment (nicolasaunai) [inline tests/simulator/test_initialization.py:605]: is this easy to get these images as artifacts in TC?

**Review by nicolasaunai — APPROVED** [review]
looks good, just 2 questions and merge (maybe squash the conflict resolution commit)

File: tests/simulator/test_initialization.py (hunk 3 above)
Comment (PhilipDeegan) [reply tests/simulator/test_initialization.py:605]: should be, so far we only export images for functional tests in nightly builds, but we can change that

File: tests/simulator/test_initialization.py (hunk 2 above)
Comment (PhilipDeegan) [reply tests/simulator/test_initialization.py:333]: moved, not added, but looks redundant too

**Pull request merged**